src/System.IO.Pipes/src/System/IO/Pipes/PipeStream.Unix.cs

```diff
@@ -255,6 +226,59 @@ public virtual PipeTransmissionMode ReadMode
         // ---- PAL layer ends here ----
         // -----------------------------
 
+        private static string s_pipeDirectoryPath;
+
+        private static string GetPipeDirectoryPath()
+        {
+            string path = s_pipeDirectoryPath;
+            if (path == null)
+            {
+                // Create the pipes temporary directory, e.g. "/tmp/.dotnet/corefx/pipes".
+                // We don't have access to Directory.CreateDirectory (which handles creating
+                // all levels of the directory full path specified), so we create each nested
+                // level individually (though we assume that the temp path already exists).
+
+                path = Path.GetTempPath();
+
+                path = Path.Combine(path, PersistedFiles.TopLevelHiddenDirectory);
+                CreateDirectory(path);
+
+                path = Path.Combine(path, PersistedFiles.SecondLevelDirectory);
+                CreateDirectory(path);
+
```

Comment:
Oh... are you too low level here to use Directory.CreateDirectory?  That makes some reluctant sense.